### PR TITLE
Remove Badger from storage client

### PR DESCRIPTION
**Pull request by aarshkshah1992**
@raulk Everything that'd interest you is in the `filestorecaradapter` package. 

### TODO
- [x] Solid Unit Tests for the FileStore CAR Adapter.

### CODE REVIEW

File: filestorecaradapter/adapter.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	rdOnly, err := blockstore.OpenReadOnly(carFilePath)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to open read-only blockstore: %w", err)
+	}
+
+	adapter := internal.BlockstoreToDSBatchingAdapter(rdOnly)
+	fm := filestore.NewFileManager(adapter, "/")
+	fm.AllowFiles = true
+	fstore := filestore.NewFilestore(rdOnly, fm)
+	bs := bstore.NewIdStore(fstore)
```
Comment (dirkmc) [inline filestorecaradapter/adapter.go:30]: Could we add some comments here to explain how this works please 🙏

**Review by dirkmc — APPROVED** [review]
Seems like it should work to me 👍

File: retrievalmarket/impl/integration_test.go [hunk 2]
```diff
@@ -362,11 +368,30 @@ func TestClientCanMakeDealWithProvider(t *testing.T) {
...
+			prepared, err := sc.Prepare()
+			require.NoError(t, err)
+			carBuf := new(bytes.Buffer)
+			require.NoError(t, prepared.Dump(carBuf))
+			require.NoError(t, fs.Close())
+			_, err = io.Copy(carFile, carBuf)
+			require.NoError(t, err)
+			_, err = carFile.Seek(0, io.SeekStart)
+			require.NoError(t, err)
+			carData, err := ioutil.ReadAll(carFile)
```
[the remark this answers is not in the document]
Comment (aarshkshah1992) [reply retrievalmarket/impl/integration_test.go:390]: @dirkmc We do need the careFile here and we write the careFile by reading from `carBuf`. However, have fixed the reading from the file again by using a `TeeReader`.

File: filestorecaradapter/internal/adapter.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	if err != nil {
+		return false, xerrors.Errorf("failed to create cid: %w", err)
+
+	}
+
+	return a.Blockstore.Has(c)
+}
+
+func (a *bsToDSBatchingAdapter) Batch() (datastore.Batch, error) {
+	return datastore.NewBasicBatch(a), nil
```
[the remark this answers is not in the document]
Comment (aarshkshah1992) [reply filestorecaradapter/internal/adapter.go:68]: This is a good question. The underlying datastore we use here (the CARv2 store) is threadsafe.
However, the batcher returned here isn't.

However, all tests right upto the end-end Lotus tests are repeatedly green. Looks like the batcher is used in a single-threaded way.

I'll write up some unit tests and run them with the race flag.

[the remark this answers is not in the document]
Comment (aarshkshah1992) [reply filestorecaradapter/internal/adapter.go:68]:
```
aarshshah@Aarshs-MacBook-Pro filestorecaradapter % go test -race -count=20 ./...
ok  	github.com/filecoin-project/go-fil-markets/filestorecaradapter	82.326s

```

**Comment by aarshkshah1992** [comment]
@dirkmc Addressed your comments and merging now so @raulk can review everything together.

**Pull request merged**